contrib: Update to libvpx 1.8.0.

bradleysepos opened the pull request:
Supersedes #1884. Smoke tested on macOS/mingw-w64. Needs encode testing.

Thanks @jstebbins for finding the pthread linking issue.

**Test on:**

- [x] Windows 10+  (via MinGW)
- [x] macOS 10.13+
- [x] Ubuntu Linux

EwoutH committed:
contrib: Update to libvpx 1.8.0.

jstebbins reviewed (approved):
Looks good.  I tested building using Fedora's mingw.  ffmpeg shows vp9 as an enabled encoder.  I didn't test running since I don't have a windows VM handy.

bradleysepos commented:
This now appears to build properly, but crashes on encode.

```
Level	Date and Time	Source	Event ID	Task Category
Information	3/1/2019 11:44:50 PM	Windows Error Reporting	1001	None	"Fault bucket , type 0
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: HandBrake.exe
P2: 1.3.0.0
P3: 5c734b28
P4: hb.DLL
P5: 0.0.0.0
P6: 00000000
P7: c0000005
P8: 000000000274970e
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERECC1.tmp.WERInternalMetadata.xml

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_HandBrake.exe_3778f571852f756a83bf14b82db0297fdb38e615_aca9d6b4_2c40e213

Analysis symbol: 
Rechecking for solution: 0
Report Id: 0070b397-5a02-4b1a-976f-17106ca234ff
Report Status: 100
Hashed bucket: 
Cab Guid: 0"
```

sr55 commented:
If you compile with symbols, you can get gdb for windows.

Otherwise, I can take a look tomorrow and see if I can identify where it's crashing.

sr55 commented:
> Encoding: task 1 of 1, 0.16 %[New Thread 4172.0x4550]
[New Thread 4172.0x4978]
[New Thread 4172.0x3b04]

Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 4172.0x3d6c]
0x00000000028aa02e in pthread_mutex_trylock ()
(gdb) bt
#0  0x00000000028aa02e in pthread_mutex_trylock ()
#1  0x00000000028aa4e8 in pthread_mutex_destroy ()
#2  0x00000000011998a4 in vp9_loop_filter_dealloc ()
#3  0x00000000011999e3 in vp9_loop_filter_frame_mt ()
#4  0x00000000010fc75b in search_filter_level ()
#5  0x00000000010fc936 in vp9_pick_filter_level ()
#6  0x00000000010f744f in encode_frame_to_data_rate ()
#7  0x00000000010f903e in vp9_get_compressed_data ()
#8  0x00000000010ea7b6 in encoder_encode ()
#9  0x00000000010c1946 in vpx_codec_encode ()
#10 0x00000000008e9ede in vpx_encode (avctx=0x36c96ac0, pkt=0x35dc9080, frame=0x4044fb00, got_packet=0x4044fabc)
    at libavcodec/libvpxenc.c:1026
#11 0x00000000007ec639 in avcodec_encode_video2 (avctx=avctx@entry=0x36c96ac0, avpkt=0x35dc9080,
    frame=frame@entry=0x4044fb00, got_packet_ptr=got_packet_ptr@entry=0x4044fabc) at libavcodec/encode.c:296
#12 0x00000000007ec9b6 in do_encode (avctx=0x36c96ac0, frame=0x4044fb00, got_packet=0x4044fabc)
    at libavcodec/encode.c:365
#13 0x00000000007ecaf3 in avcodec_send_frame (avctx=0x36c96ac0, frame=frame@entry=0x4044fb00)
    at libavcodec/encode.c:414
#14 0x000000000045b64d in Encode (w=w@entry=0x35d83c60, in=in@entry=0x37f66f30, list=list@entry=0x4044fd80)
    at ../libhb/encavcodec.c:906
#15 0x000000000045b756 in encavcodecWork (w=0x35d83c60, buf_in=<optimized out>, buf_out=0x4044fdf8)
    at ../libhb/encavcodec.c:967
#16 0x000000000047678c in hb_work_loop (_w=0x35d83c60) at ../libhb/work.c:1931
#17 0x0000000000444a2f in hb_thread_func (_t=0x36a933a0) at ../libhb/ports.c:867
#18 0x00000000028acca1 in ptw32_threadStart ()
#19 0x00007ffcf165b0ba in msvcrt!_beginthreadex () from C:\WINDOWS\System32\msvcrt.dll
#20 0x00007ffcf165b18c in msvcrt!_endthreadex () from C:\WINDOWS\System32\msvcrt.dll
#21 0x00007ffcf0af81f4 in KERNEL32!BaseThreadInitThunk () from C:\WINDOWS\System32\kernel32.dll
#22 0x00007ffcf369a251 in ntdll!RtlUserThreadStart () from C:\WINDOWS\SYSTEM32\ntdll.dll
---Type <return> to continue, or q <return> to quit---
#23 0x0000000000000000 in ?? ()
Backtrace stopped: previous frame inner to this frame (corrupt stack?)
(gdb)

bradleysepos commented:
GDB

```
Thread 47 received signal SIGSEGV, Segmentation fault.
[Switching to Thread 3520.0x1df0]
0x000000000264ce2e in ?? ()
(gdb) bt
#0  0x000000000264ce2e in ?? ()
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
```

VS

```
'HandBrake.exe' (CLR v4.0.30319: DefaultDomain): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_64\mscorlib\v4.0_4.0.0.0__b77a5c561934e089\mscorlib.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: DefaultDomain): Loaded 'C:\Users\Bradley\Desktop\HandBrake\win\CS\HandBrakeWPF\bin\x64\Debug\HandBrake.exe'. Symbols loaded.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework\v4.0_4.0.0.0__31bf3856ad364e35\PresentationFramework.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\WindowsBase\v4.0_4.0.0.0__31bf3856ad364e35\WindowsBase.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Core\v4.0_4.0.0.0__b77a5c561934e089\System.Core.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System\v4.0_4.0.0.0__b77a5c561934e089\System.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_64\PresentationCore\v4.0_4.0.0.0__31bf3856ad364e35\PresentationCore.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Xaml\v4.0_4.0.0.0__b77a5c561934e089\System.Xaml.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\Remote Debugger\x64\Runtime\Microsoft.VisualStudio.Debugger.Runtime.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Configuration\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Configuration.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Xml\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework.Aero2\v4.0_4.0.0.0__31bf3856ad364e35\PresentationFramework.Aero2.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\Users\Bradley\Desktop\HandBrake\win\CS\HandBrakeWPF\bin\x64\Debug\Caliburn.Micro.Platform.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\Users\Bradley\Desktop\HandBrake\win\CS\HandBrakeWPF\bin\x64\Debug\Caliburn.Micro.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Runtime\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Runtime.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Threading.Tasks\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Threading.Tasks.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\Users\Bradley\Desktop\HandBrake\win\CS\HandBrakeWPF\bin\x64\Debug\Caliburn.Micro.Platform.Core.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Reflection\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Reflection.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Collections\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Collections.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.ObjectModel\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.ObjectModel.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Linq\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Linq.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\Users\Bradley\Desktop\HandBrake\win\CS\HandBrakeWPF\bin\x64\Debug\HandBrake.Interop.dll'. Symbols loaded.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\Users\Bradley\Desktop\HandBrake\win\CS\HandBrakeWPF\bin\x64\Debug\System.Windows.Interactivity.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Reflection.Extensions\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Reflection.Extensions.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\Users\Bradley\Desktop\HandBrake\win\CS\HandBrakeWPF\bin\x64\Debug\Newtonsoft.Json.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Numerics\v4.0_4.0.0.0__b77a5c561934e089\System.Numerics.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Runtime.Serialization\v4.0_4.0.0.0__b77a5c561934e089\System.Runtime.Serialization.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_64\System.Data\v4.0_4.0.0.0__b77a5c561934e089\System.Data.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'Microsoft.GeneratedCode'. 
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'Microsoft.GeneratedCode'. 
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'Microsoft.GeneratedCode'. 
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Linq.Expressions\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Linq.Expressions.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Drawing\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Drawing.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Windows.Forms\v4.0_4.0.0.0__b77a5c561934e089\System.Windows.Forms.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Management\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Management.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
The thread 0x1e0c has exited with code 0 (0x0).
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Text.RegularExpressions\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Text.RegularExpressions.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Threading\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Threading.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Resources.ResourceManager\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Resources.ResourceManager.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
System.Windows.Data Error: 1 : Cannot create default converter to perform 'one-way' conversions between types 'System.Windows.Controls.ContextMenu' and 'System.Windows.Rect'. Consider using Converter property of Binding. BindingExpression:Path=; DataItem='ContextMenu' (Name='MainContextMenu'); target element is 'ContextMenu' (Name='MainContextMenu'); target property is 'PlacementRectangle' (type 'Rect')
System.Windows.Data Error: 5 : Value produced by BindingExpression is not valid for target property.; Value='System.Windows.Controls.ContextMenu Items.Count:0' BindingExpression:Path=; DataItem='ContextMenu' (Name='MainContextMenu'); target element is 'ContextMenu' (Name='MainContextMenu'); target property is 'PlacementRectangle' (type 'Rect')
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\Users\Bradley\Desktop\HandBrake\win\CS\HandBrakeWPF\bin\x64\Debug\GongSolutions.WPF.DragDrop.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
System.Windows.Data Error: 1 : Cannot create default converter to perform 'one-way' conversions between types 'System.Windows.Controls.ContextMenu' and 'System.Windows.Rect'. Consider using Converter property of Binding. BindingExpression:Path=; DataItem='ContextMenu' (Name='MainContextMenu'); target element is 'ContextMenu' (Name='MainContextMenu'); target property is 'PlacementRectangle' (type 'Rect')
System.Windows.Data Error: 5 : Value produced by BindingExpression is not valid for target property.; Value='System.Windows.Controls.ContextMenu Items.Count:0' BindingExpression:Path=; DataItem='ContextMenu' (Name='MainContextMenu'); target element is 'ContextMenu' (Name='MainContextMenu'); target property is 'PlacementRectangle' (type 'Rect')
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework-SystemXml\v4.0_4.0.0.0__b77a5c561934e089\PresentationFramework-SystemXml.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\UIAutomationTypes\v4.0_4.0.0.0__31bf3856ad364e35\UIAutomationTypes.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework-SystemData\v4.0_4.0.0.0__b77a5c561934e089\PresentationFramework-SystemData.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\Users\Bradley\AppData\Local\Temp\VisualStudio.XamlDiagnostics.13208\Microsoft.VisualStudio.DesignTools.WpfTap.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\SMDiagnostics\v4.0_4.0.0.0__b77a5c561934e089\SMDiagnostics.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.ServiceModel.Internals\v4.0_4.0.0.0__31bf3856ad364e35\System.ServiceModel.Internals.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\UIAutomationProvider\v4.0_4.0.0.0__31bf3856ad364e35\UIAutomationProvider.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\Accessibility\v4.0_4.0.0.0__b03f5f7f11d50a3a\Accessibility.dll'. Cannot find or open the PDB file.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Configuration.Install\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Configuration.Install.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'HandBrake.exe' (CLR v4.0.30319: HandBrake.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Runtime.Serialization.Primitives\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Runtime.Serialization.Primitives.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
The thread 0xbdc has exited with code 0 (0x0).
The program '[13208] HandBrake.exe' has exited with code -1073741819 (0xc0000005) 'Access violation'.
```

bradleysepos commented:
Note sure why GDB isn't giving me more info. I configured with `--debug=max` and `hb.dll` is huge.

Anyway, it looks like they may have broken compatibility with pthreadsw32.

bradleysepos commented:
https://git.ffmpeg.org/gitweb/ffmpeg.git/commitdiff/402bf262375dfecd0e90d7acc67c238abe952fc3

bradleysepos committed:
contrib: Remove local pthreads-w32 in favor of winpthreads.

Cross-compiling for Windows now requires winpthreads which is part of the toolchain provided by scripts/mingw-w64-build, and similar packages provided by Linux distros.

bradleysepos committed:
Update to mingw-w64-build 6.0.0.

Adds winpthreads and gdb. Disable gdb with --disable-gdb.

bradleysepos commented:
I've revised this PR to update `mingw-w64-build` with winpthreads support (also available as a mingw-w64 addon package on most Linux distros), remove pthreads-w32 in favor of this threading library, and finally incorporate the remaining libvpx fixes.

Tested encoding VP9 and x264 on Windows. Seems to work like a charm. Additional testing is welcome, as swapping the threading library is not necessarily a trivial change.

bradleysepos commented:
For posterity, this seems to indicate libvpx no longer supports pthreads-w32, which current HandBrake uses, and this PR addresses by switching to winpthreads: https://github.com/webmproject/libvpx/commit/38dc27cc6d03383cf1764e03697137e2f330d550

bradleysepos commented:
@sr55 Any objection to merging?

jstebbins commented:
Fine with me.

bradleysepos committed:
scripts: Update to mingw-w64-build 6.0.1.

Fixes an issue with GDB makeinfo path.

sr55 commented:
Tested ok for VP8/9 here now.

Tar uploaded!
